operate/client/src/App/ProcessInstance/FlowNodeInstanceLog/FlowNodeInstancesTree/v2/elementInstancesTreeStore.ts

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+
+import type {ElementInstance} from '@camunda/camunda-api-zod-schemas/8.8';
+import {makeAutoObservable} from 'mobx';
+import {searchElementInstances} from 'modules/api/v2/elementInstances/searchElementInstances';
+import {logger} from 'modules/logger';
+
+const PAGE_SIZE = 50;
+
+type PageMetadata = {
+  totalItems: number;
+  windowStart: number;
+  windowEnd: number;
+};
+
+type NodeData = {
+  items: ElementInstance[];
+  pageMetadata: PageMetadata;
+  status: 'idle' | 'loading' | 'error' | 'loaded';
+};
+
+type State = {
+  rootScopeKey: string | null;
+  nodes: Map<string, NodeData>;
+  expandedNodes: Set<string>;
+  abortControllers: Map<string, AbortController>;
+};
+
+class ElementInstancesTreeStore {
+  state: State = {
+    rootScopeKey: null,
+    nodes: new Map(),
+    expandedNodes: new Set(),
+    abortControllers: new Map(),
+  };
+
+  constructor() {
+    makeAutoObservable(this);
+  }
+
+  private getOrCreateAbortController = (scopeKey: string): AbortController => {
+    const existing = this.state.abortControllers.get(scopeKey);
+
+    if (existing && !existing.signal.aborted) {
+      return existing;
+    }
+
+    const controller = new AbortController();
+    this.state.abortControllers.set(scopeKey, controller);
+    return controller;
+  };
+
+  setRootNode = async (processInstanceKey: string) => {
+    if (this.state.rootScopeKey !== processInstanceKey) {
```

Comment:
🔧 Using a proper early return makes the early return easier to spot and reduced indentation.

```ts
if (this.state.rootScopeKey === processInstanceKey) return;
```